code/iaas/simple-allocator/src/main/java/io/cattle/platform/simple/allocator/dao/impl/SimpleAllocatorDaoImpl.java

```diff
@@ -199,5 +212,48 @@ public ObjectManager getObjectManager() {
     public void setObjectManager(ObjectManager objectManager) {
         this.objectManager = objectManager;
     }
+    
+    
+    @Override
+    public ResourceRequest populateResourceRequestFromInstance(Instance instance, String resourceType, String poolType) {
```

Comment:
If you want to encapsulate all this logic in a method, that is fine, but a Dao method is not the right place. Dao methods should deal solely with retrieving data from the DB and have very little knowledge of the system/business logic outside of that.

You could move a method like this to simpleAllocator